atomisticparsers/h5md/parser.py

```diff
@@ -0,0 +1,690 @@
+    def parse_method(self):
+
+        sec_method = Method()
+        self.archive.run[-1].method.append(sec_method)
+        sec_force_field = sec_method.m_create(ForceField)  # TODO @landinesa how do we remove m_create's here?
+        sec_model = sec_force_field.m_create(Model)
+
+        # get the atom parameters
+        n_atoms = self._n_atoms[0] if self._n_atoms is not None else 0  # TODO Extend to non-static n_atoms
+        for n in range(n_atoms):
+            sec_atom = AtomParameters()
+            sec_method.atom_parameters.append(sec_atom)
+
+            for key in self._atom_parameters.keys():
+                sec_atom.m_set(sec_atom.m_get_quantity_definition(key), self._atom_parameters[key][n])
+
+        # Get the interactions
+        connectivity_group = self._data_parser.get('connectivity')
+        if connectivity_group:
+            atom_labels = self._atom_parameters.get('label')
+            interaction_keys = ['bonds', 'angles', 'dihedrals', 'impropers']
+            interactions_by_type = []
+            for interaction_key in interaction_keys:
+                interaction_list = self._data_parser.get(f'connectivity.{interaction_key}')
+                if interaction_list is None:
+                    continue
+                elif isinstance(interaction_list, h5py.Group):
+                    self.logger.warning('Time-dependent interactions currently not supported. These values will not be stored')
+                    continue
+
+                interaction_type_dict = {
+                    'type': interaction_key,
+                    'n_interactions': len(interaction_list),
+                    'n_atoms': len(interaction_list[0]),
+                    'atom_indices': interaction_list,
+                    'atom_labels': np.array(atom_labels)[interaction_list] if atom_labels is not None else None
+                }
+                interactions_by_type.append(interaction_type_dict)
+            self.parse_interactions_by_type(interactions_by_type, sec_model)
+
+        # Get the force calculation parameters
+        force_calculation_parameters = self._parameter_info.get('force_calculations')
+        if force_calculation_parameters:
+            sec_force_calculations = sec_force_field.m_create(ForceCalculations)
+            sec_neighbor_searching = sec_force_calculations.m_create(NeighborSearching)
+
+            for key, val in force_calculation_parameters.items():
+                if not isinstance(val, dict):
+                    if self.is_valid_key_val(ForceCalculations, key, val):
+                        sec_force_calculations.m_set(sec_force_calculations.m_get_quantity_definition(key), val)
+                    else:
+                        units = val.units if hasattr(val, 'units') else None
+                        val = val.magnitude if units is not None else val
+                        sec_force_calculations.x_h5md_parameters.append(ParamEntry(kind=key, value=val, unit=units))
+                else:
```

Comment:
the else here is not necessary use: elif key == 'neighbor_searching' else:

Reply:
done.